e2e/test_reconnection.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+set -e
+
+case "$1" in
+  rest)
+    rest_arg="--rest"
+    server_app_address="http://localhost:9091"
+    server_address="http://localhost:9093"
+    db_arg="--database :flwr-in-memory-state:"
+    ;;
+  sqlite)
+    rest_arg=""
+    server_address="127.0.0.1:9092"
+    server_app_address="127.0.0.1:9091"
+    db_arg="--database $(date +%s).db"
+    ;;
+  *)
+    rest_arg=""
+    server_address="127.0.0.1:9092"
+    server_app_address="127.0.0.1:9091"
+    db_arg="--database :flwr-in-memory-state:"
+    ;;
+esac
+
+dir_arg="--dir ./.."
+
+timeout 2m flower-superlink --insecure $db_arg $rest_arg &
+sl_pid=$!
+sleep 3
+
+timeout 2m flower-client-app client:app --insecure $rest_arg --server $server_address &
+cl1_pid=$!
+sleep 3
+
+timeout 2m flower-client-app client:app --insecure $rest_arg --server $server_address &
+cl2_pid=$!
+sleep 3
+
+# Kill superlink
+kill $sl_pid
+sleep 3
+
+# Restart superlink
+timeout 2m flower-superlink --insecure $db_arg $rest_arg &
+sl_pid=$!
+sleep 20
+
+# Kill first client
+kill $cl1_pid
+sleep 3
+
+# Restart first client
+timeout 2m flower-client-app client:app --insecure $rest_arg --server $server_address &
+cl1_pid=$!
+sleep 5
+
+timeout 2m flower-server-app server:app --insecure $dir_arg $rest_arg --server $server_app_address &
+pid=$!
+
+# Kill first client
+kill $cl1_pid
+sleep 1
+
+# Restart first client
+timeout 2m flower-client-app client:app --insecure $rest_arg --server $server_address &
+cl1_pid=$!
+
+wait $pid
+res=$?
+
+if [[ "$res" = "0" ]];
```

Comment:
I wonder if it would it be better to use [`bats`](https://github.com/bats-core/bats-core) to test this shell script rather than relying on the exit statement? There's an example in their CI [here](https://github.com/bats-core/bats-core/blob/master/.github/workflows/tests.yml#L200-L212).

Reply:
I didn't know about this library, it looks interesting! I think this should be left for another PR though (as it would apply to many places in our codebase, and we might also ditch bash for most of our tests)